.github/workflows/publish.yml

```diff
@@ -37,6 +37,10 @@ jobs:
           GH_TOKEN: ${{ github.token }}
         shell: pwsh
         run: |
+          if ( -Not ( $env:RELEASE_TAG -match '^v3\.[0-9]+\.[0-9]+(-preview)*$' ) ){
```

Comment:
We should allow extra characters after `-preview`. There may be more than one preview releases for each version number.